src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesLegendURLTest.java

```diff
@@ -404,8 +404,37 @@ public void testOnlineResourceWidthHeight() throws Exception {
         Element onlineResource = (Element) onlineResources.item(0);
         String href = onlineResource.getAttribute("xlink:href");
         assertNotNull(href);
-        assertTrue(href.contains("width=20"));
-        assertTrue(href.contains("height=20"));
+        assertTrue(href.contains("width=50"));
+        assertTrue(href.contains("height=10"));
+    }
+
+    /**
+     * Tests that width and height in legend online resources are consistent with the ones specified
+     * in legendURL and not always with default value of 20.
+     */
+    @Test
+    public void testWidthHeightInHrefConsistentWithLegendURL() throws Exception {
+        TransformerBase tr = createTransformer();
+        tr.setIndentation(2);
+        Document dom = WMSTestSupport.transform(req, tr);
+
+        NodeList legendURLS = XPATH.getMatchingNodes(getLegendURLXPath("cite:Bridges"), dom);
```

Comment:
This one also has custom legend? clarify

Reply:
fixed